packages/manager/src/components/EnhancedSelect/variants/RegionSelect/RegionSelect.tsx

```diff
@@ -46,11 +51,30 @@ export const getRegionOptions = (regions: Region[]) => {
     Other: [],
   };
 
-  for (const region of regions) {
+  const hasUserAccessToDisabledRegions = listOfDisabledRegions.some(
+    (disabledRegion) =>
+      regions.some((region) => region.id === disabledRegion.regionId)
+  );
+  const allRegions = [
+    ...regions,
+    ...listOfDisabledRegions
+      .filter((disabledRegion) => flags[disabledRegion.featureFlag] === true)
+      .map((region) => region.fakeRegion)
+      .filter(
+        (fakeRegion) => !regions.some((region) => region.id === fakeRegion.id)
+      ),
```

Comment:
Minor (feel free to consider as a follow-on): I think you could combine the filters, drop the `=== true`, and make the variable names more consistent. Something like: 

```suggestion
      .filter(
        (disabledRegion) =>
          flags[disabledRegion.featureFlag] &&
          // Don't display a fake region if it's included in the real /regions response
          !regions.some((region) => region.id === disabledRegion.fakeRegion.id)
      )
      .map((disabledRegion) => disabledRegion.fakeRegion),
```